src/main/controllers/tx-history-service.js

```diff
@@ -0,0 +1,276 @@
+'use strict';
+
+const Promise = require('bluebird'),
+    electron = require('electron'),
+    path = require('path'),
+    config = require('../config'),
+    request = require('request'),
+    async = require('async'),
+    BigNumber = require('bignumber.js');
+
+let isSyncingMap = {};
+let syncingJobIsStarted = false;
+
+let getIsSyncing = (address) => {
+    if (!address) {
+        return false;
+    }
+
+    return isSyncingMap[address];
+};
+
+let defaultModule = function (app) {
+    const API_KEY = null;
+    const REQUEST_INTERVAL_DELAY = 600; // millis
+    const LAST_BLOCK = Number.MAX_SAFE_INTEGER;
+    const ETH_BALANCE_DIVIDER = new BigNumber(10 ** 18);
+    const ENDPOINT_CONFIG = {
+        1: { url: 'https://api.etherscan.io/api' },
+        3: { url: "http://api-ropsten.etherscan.io/api" }
+    };
+    const API_ENDPOINT = ENDPOINT_CONFIG[config.chainId].url;
+
+    let OFFSET = 1000;
+
+    const TX_LIST_ACTION = `?module=account&action=txlist&startblock=0&sort=asc&offset=${OFFSET}`;
+    const TOKEN_TX_ACTION = `?module=account&action=tokentx&startblock=0&sort=asc&offset=${OFFSET}`;
+    const TX_RECEIPT_ACTION = '?module=proxy&action=eth_getTransactionReceipt';
+
+
+    //in order to change key name in runtime
+    const KEY_MAP = {
+        'txreceipt_status': 'txReceiptStatus'
+    };
+
+    const KNOWN_KEYS = ['blockNumber', 'timeStamp', 'hash', 'nonce', 'blockHash',
+        'transactionIndex', 'from', 'to', 'value', 'gas', 'gasPrice', 'isError',
+        'txreceipt_status', 'input', 'contractAddress', 'cumulativeGasUsed',
+        'gasUsed', 'confirmations', 'tokenName', 'tokenSymbol', 'tokenDecimal'];
+
+    let queue = async.queue((args, callback) => {
+        let result = makeRequest.apply(this, [args.method, args.url, args.data])
+        setTimeout(() => {
+            callback(result);
+        }, REQUEST_INTERVAL_DELAY);
+    }, 1);
+
+    function loadEthTxHistory(address, startblock, endblock, page) {
+        return new Promise((resolve, reject) => {
+            const ACTION_URL = `${API_ENDPOINT}${TX_LIST_ACTION}&address=${address}&startblock=${startblock}&endblock=${endblock}&page=${page}`;
+            queue.push({ method: 'get', url: ACTION_URL }, (promise) => {
+                resolve(promise);
+            });
+        });
+    }
+
+    async function loadERCTxHistory(address, startblock, endblock, page) {
+        return new Promise((resolve, reject) => {
+            const ACTION_URL = `${API_ENDPOINT}${TOKEN_TX_ACTION}&address=${address}&startblock=${startblock}&endblock=${endblock}&page=${page}`;
+            queue.push({ method: 'get', url: ACTION_URL }, (promise) => {
```

Comment:
In general I think repeating code this much is not that good, could you possibly consider creating some kind ``request`` function?